docs/source/gbd2017_models/concept_models/vivarium_conic_lsff/concept_model.rst

```diff
@@ -1391,25 +1403,39 @@ Summary of Iron Intervention Algorithm
   mother_iron_group   = ('baseline' if mother_fortified_baseline_i else
                          'intervention_not_baseline' if mother_fortified_i else
                          'uncovered')
-  bw_shift_i = bw_response * iron_concentration * daily_flour_i
-  birth_weight_i      = birthweight_gbd_i + (
-                        -baseline_coverage(t) + mother_fortified_i
-                        ) * bw_shift_i
-
-  # At each simulation time step (including when simulant is born)
-  child_fortified_i(t)  = time_covered_i <= t
-                        = p_i < coverage(t)
-    # If child is covered in baseline, then time_covered_i = float('-inf'),
-    # and hence hb_lag_fraction(t-time_covered_i) = 1, i.e. the child receives
-    # the full age-dependent effect in this case since the time lag has passed.
-  hb_i(t)               = hb_gbd(age_i(t)) + (
-                          -baseline_coverage(t) + child_fortified_i(t) * hb_lag_fraction(t-time_covered_i)
-                          ) * hb_shift *  hb_age_fraction(age_i(t))
+    # Effect on birthweight
+  daily_flour_i       = average daily flour consumption Y of simulant's mother
+                      = (flour_consumption_dist.rvs() if mother_fortified_i else
+                         'irrelevant')
+  bw_shift_i          = birthweight shift Z given flour consumption Y (if mother was fortified)
+                      = bw_response * iron_concentration * daily_flour_i
+  birth_weight_i      = (birthweight_gbd_i
+                         - baseline_coverage(t) * mean_bw_shift
+                         + mother_fortified_i * bw_shift_i)
+
+  ## At each simulation time step (including t=0 or when simulant is born):
+  household_covered_i(t)    = time_covered_i <= t
+                            = p_i < coverage(t)
+    # Fortification starts either when the household receives coverage, or when
+    # the child turns 6 months old and starts eating solids, whichever is later.
+    # Recall that if the child is covered in baseline, then time_covered_i =
+    # float('-inf'), so coverage always starts at age 6 months in this case.
+  time_since_fortified_i(t) = min(t - time_covered_i, t - age(t) + 0.5)
```

Comment:
I'm confused about why we are taking the minimum of the time coverage and age components here for `time_since_fortified_i(t) = min(t - time_covered_i, t - age(t) + 0.5)`. My thought was that the `hb_lag_fraction(time_since_fortified_i(t))` should only be based on time since coverage and then the `hb_age_fraction(age_i(t))` takes care of the age component. Let me know if I'm misunderstanding what's going on here.

Reply:
Well, I think that's how we were originally thinking about it, but after thinking about it some more, it seems that our original approach is slightly inconsistent with points 1-4 in the section [Determining whether a simulant is affected -Iron](https://vivarium-research.readthedocs.io/en/latest/gbd2017_models/concept_models/vivarium_conic_lsff/concept_model.html#determining-whether-a-simulant-is-affected-iron), and point 5 in that section is also slightly inconsistent with the other 4. Here's my reasoning:

We are trying to capture two effects:
1. Children don't start eating solid food until 6 months, at which point the amount of food slowly increases to a maximum at 2 years.
2. When someone starts eating iron-fortified food, the measured effect on hemoglobin increases from 0 to the maximum effect size over a period of 6 months.

The function `hb_age_fraction(age_i(t))` captures the first effect, while the function `hb_lag_fraction(time_since_fortified_i(t))` captures the second effect.

Now suppose we use our original strategy and set `time_since_fortified_i(t) = t - time_covered_i`, regardless of age, and suppose `time_covered_i` is before the simulant is born. Then we'd have `hb_lag_fraction(time_since_fortified_i(t))=1` when the child is six months old, which means we'd be saying that by age 6 months the simulant would already have been eating fortified food long enough to see the full effect (which happens to be 0 because of the age function, but that's beside the point), _even though they have not eaten any fortified food at that age!_ What I'm saying is that `time_since_fortified_i(t)` should be the actual amount of time you've been _eating_ fortified food, not just the time since you're household started receiving the food, because `hb_lag_fraction(time_since_fortified_i(t))` is supposed to capture a biological effect that depends on when you started eating the food.

Does that make sense?

Mathematically, the result is that for children whose household gets covered before they're  6 months old, their effect size will increase quadratically instead of linearly at first (since we're multiplying two linear functions), with slope 0 at 6 months, then curving upward until it hits the blue line at 1 year (1/3 the maximum effect size), then follows the blue graph from there.  I think slope 0 at 6 months seems more reasonable than positive slope at 6 months, though the difference in our results should be negligible.  Also, this is the same as what would have happened already if coverage is received at 6 months instead of birth. So basically I'm just saying there should be no difference between receiving coverage at 6 months and receiving coverage at birth since the child is not eating food during that time.

Reply:
Ah, ok... this is altering our assumptions slightly then I think (or at least the implicit assumptions that I made but maybe we never explicitly stated). My That sounds fine to me although we should definitely write that down in prose. I think that we had initially settled on the linear curve because we don't really know what happens in this situation and this was we were airing on the side of not being overly conservative (so that we can demonstrate that this will have minimal impact even without conservative constraints) as Abie had mentioned a few times. I don't really have a preference except I think that we need to clear up some of the contradictions between this section and the info in this screenshot. 
![image](https://user-images.githubusercontent.com/54918472/83204905-e2c7b700-a101-11ea-8629-32238cb2fd3a.png)

Reply:
Ahh, I think I mistranslated your code when I split the age part and time lag part into separate functions.  I didn't quite understand what you were doing with the `age_of_coverage_i` variable, so when I reworked it, I wrote the `hb_age_fraction` function in terms of the simulant's current age.

But that has the effect of making `hb_age_fraction(age(t))` a linear function of t, whereas I think your variable `age_of_coverage_i` is a constant. So in my version, I am multiplying two linear functions together when `0.5<age(t)<2`, giving a quadratic function, whereas in your version you're multiplying a linear function by a constant, giving a linear function. (You'd think with a math PhD, I would have noticed this difference sooner, but I just noticed it yesterday...)

I think there were two reasons I was trying to split up the Hb shift into two pieces in the first place:

1. To make it easier to understand what's going on, and
2. Because for baseline calibration, I _think_ it makes sense to apply only the age part to the downshift but to apply both the age part and the time lag part to the upshift. (This will make our mean Hb slightly lower than GBD's, but it seems more logical than applying the time lag to the downshift as well.)

Does that second point seem correct to you, @alibow ? If so, I'm not sure how do do the baseline calibration correctly while preserving the overall linear scaleup in your version.

Reply:
Ah, I see what you're saying... essentially it doesn't make sense to have the down shift be time-dependent, which is what happens in my version, but not yours. Is that the idea? If so then I think I agree with your approach.

Reply:
Yeah, I think that's the issue (though it's still a bit confusing for me to think about). Let's go with my version then, since it gives us a reasonable approach for baseline calibration.  I think I'll try drawing some graphs of both your function and my function to make sure everything looks how I think it should. Is your code for drawing the above graph on the vivarium_data_analysis repo?

Reply:
@KjellSwedin What's relevant for you here is that the research decision is to go with the version of the Hb shift (including baseline calibration) that's in the "Summary of Iron Intervention Algorithm" section in this PR, and *not* the version Ali posted above in the "Determining Whether A Simulant is Affected - Iron" section, which is slightly different and doesn't handle baseline calibration. Does that make sense?

I'm about to start updating the other sections to be consistent with the new version here. Let me know if you have any questions.